src/two_party/lindell_2017/party2.rs

```diff
@@ -225,17 +228,35 @@ impl MasterKey2 {
                     party_two_paillier,
                     challenge,
                     verification_aid,
+                    pdl_chal,
                 ))
             }
-            Err(_range_proof_error) => Err(ProofError),
+            Err(_range_proof_error) => Err(()),
         }
     }
 
     pub fn key_gen_third_message(
         proof_result: &CorrectKeyProof,
         verification_aid: &VerificationAid,
-    ) -> Result<(), CorrectKeyProofError> {
-        party_two::PaillierPublic::verify_correct_key(proof_result, verification_aid)
+        pdl_chal: &party_two::PDLchallenge,
+    ) -> Result<(party_two::PDLdecommit), ()> {
```

Comment:
Also you remove the Error type here, any reason? Can we create a new error type if ProofError do not fit here?

Reply:
see above comment

Reply:
Ok see above